spec/mem/tree_borrows/memory.md

```diff
@@ -138,18 +138,28 @@ impl<T: Target> TreeBorrowsMemory<T> {
     fn release_protector(&mut self, provenance: TreeBorrowsProvenance) -> Result {
         let (alloc_id, path) = provenance;
         self.mem.allocations.mutate_at(alloc_id.0, |allocation| {
-            let protected_node = allocation.extra.root.get_node(path);
-            assert_ne!(protected_node.protected, Protected::No);
+            let protector_accesses : List<_> = allocation.extra.root.access_node(path, |node| {
+                assert!(node.protected.yes());
+                // we would like to iter_mut here, but sadly this is not supported in specr.
+                let mut new_perms = list![];
+                let mut derived_accs = list![];
+                for mut permission in node.permissions {
+                    let derived_access = permission.unprotect();
```

Comment:
This doesn't look like the `unprotect` method with its `&mut` is the right signature for this caller.